subprojects/core/src/main/java/org/gradle/internal/work/DefaultAsyncWorkTracker.java

```diff
@@ -92,6 +92,22 @@ public void run() {
         }
     }
 
+    @Override
+    public boolean hasUncompletedWork(BuildOperationRef operation) {
+        lock.lock();
+        try {
+            List<AsyncWorkCompletion> workItems = ImmutableList.copyOf(items.get(operation));
```

Comment:
⭕️ Why the copy?

Reply:
Bleh, cut/paste error.  I've removed the copy.